src/providers/CartContext.tsx

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+
+import { AppProviderProps } from './AppProvider';
+
+const CartContext = React.createContext({});
+
+const CartProviderComponent = ({ children }: AppProviderProps) => {
```

Comment:
Same here, you can just do:
```suggestion
const CartProviderComponent: React.FC = ({ children }) => {
```
------

and if you need to extend it you can do:
```suggestion
interface CartProviderComponentProps {
  someProp: bool
}
const CartProviderComponent: React.FC<CartProviderComponentProps> = ({ children }) => {
```
-----

This way you don't recreate something that React already provides and you avoid coupling of these context with the AppProvider file.

Reply:
I wasn't so sure about this, thanks for simplifying.

Reply:
I attempted this but got these errors.

```
ERROR in src/providers/AppProvider.tsx:12:6
TS2559: Type '{ children: Element; }' has no properties in common with type 'IntrinsicAttributes'.
    10 | export const AppProvider: React.FC<AppProviderProps> = ({ children }) => (
    11 |   <ProductsProvider>
  > 12 |     <CartProvider>
       |      ^^^^^^^^^^^^
    13 |       <SessionProvider>
    14 |         <BrowserRouter>{children}</BrowserRouter>
    15 |       </SessionProvider>

ERROR in src/providers/CartContext.tsx:5:44
TS2339: Property 'children' does not exist on type '{}'.
    3 | const CartContext = React.createContext({});
    4 |
  > 5 | const CartProviderComponent: React.FC = ({ children }) => {
      |                                            ^^^^^^^^
    6 |   const cart = {};
    7 |
    8 |   const addToCart = () => {};

```

 I've done a bit of research and apparently I have to explicitly pass the props type including the `children` like this:
 ```
 type CartProviderProps = { children: React.ReactNode };

const CartProviderComponent: React.FC<CartProviderProps> = ({ children }) => {
```
 
Seems like the [`FC` type definition](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/9bdc01d75d8e9d3fb651a63b9e94547a01b228ad/types/react/index.d.ts#L515-L523) contains the `children` key, but the default type is an empty object.

Reply:
I think I'll create a types file to handle all these common interfaces to avoid coupling and circular dependencies

Reply:
This might be what's happening: [In React 18 you now have to explicitly declare children as props if you want a component to have it](https://reactjs.org/blog/2022/03/08/react-18-upgrade-guide.html#updates-to-typescript-definitions).

So my comment here about using React.FC to avoid declaring children can be dismissed. I will re-review the circular dependency now, since I saw you made changes to that